src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -32,56 +32,82 @@ namespace Microsoft.Diagnostics.Tracing.Analysis
     /// </summary>
     public static class TraceLoadedDotNetRuntimeExtensions
     {
+        internal static readonly string LoadedDotNetRuntimesKey = "Computers/LoadedDotNetRuntimes";
+
         public static void NeedLoadedDotNetRuntimes(this TraceEventDispatcher source)
         {
+            if (source.userData.ContainsKey(LoadedDotNetRuntimesKey))
+            {
+                // The callbacks are already registered for this source
+                return;
+            }
+
             // ensure there are base processes
```

Comment:
❓ Can you describe the method by which you expect this information to be cleared in order to restart processing using the same source but without the previously accumulated data?